ledger-service/http-json/src/main/scala/com/digitalasset/http/query/ValuePredicate.scala

```diff
@@ -296,21 +302,19 @@ object ValuePredicate {
     }
 
     def fromVariant(
-        fields: Map[String, JsValue],
+        tag: String,
+        nestedValue: JsValue,
         id: Ref.Identifier,
-        fieldTyps: ImmArraySeq[(Ref.Name, Ty)]): Result = fields.toSeq match {
-      case Seq((k, v)) =>
-        val name = Ref.Name.assertFromString(k)
-        val field: Option[(Ref.Name, Ty)] = fieldTyps.find(_._1 == name)
-        val fieldP: Option[(Ref.Name, ValuePredicate)] = field.map {
-          case (n, t) => (n, fromValue(v, t))
-        }
-        fieldP.fold(
-          predicateParseError(
-            s"Cannot locate Variant's (datacon, type) field, id: $id, name: $name")
-        )(VariantMatch)
+        fieldTyps: ImmArraySeq[(Ref.Name, Ty)]): Result = {
 
-      case _ => predicateParseError(s"Variant must have exactly 1 field, got: $fields, id: $id")
+      val name = Ref.Name.assertFromString(tag)
+      val field: Option[(Ref.Name, Ty)] = fieldTyps.find(_._1 == name)
```

Comment:
```suggestion
      val field: Option[(Ref.Name, Ty)] = fieldTyps.find { case (name, _) => tag == (name: String) }
```

and delete `val name`.

Reply:
The following `map` also fuses with that `collect`, so `val field` should also be deleted.